pegen/pegen.h

```diff
@@ -54,11 +54,31 @@ void *keyword_token(Parser *p, const char *val);
 
 void *CONSTRUCTOR(Parser *p, ...);
 
+#define expr_type_headline(a) ((expr_ty) a)->lineno
```

Comment:
I wonder if these should be made real functions after all. We can then make the argument types `expr_ty` etc. which would catch mistakes where something that's known to be e.g. `stmt_ty` will be flagged as an error. `void *` will still work. The functions can be made `inline` for performance.

Reply:
Done! I was not sure, if it's okay to put these directly in the header file, so let me know if it is.